src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1560,6 +1566,12 @@ private BoundExpression InferTypeForDiscardAssignment(BoundDiscardExpression op1
                                 op2 = ToBadExpression(op2);
                             }
                         }
+                        else if (op1.Kind is BoundKind.Local or BoundKind.Parameter &&
+                            GetValEscape(op1, LocalScopeDepth) != GetValEscape(op2, LocalScopeDepth))
+                        {
+                            Error(diagnostics, ErrorCode.ERR_RefAssignNarrower, node, getName(op1), op2.Syntax);
```

Comment:
>ErrorCode.ERR_RefAssignNarrower [](http://example.com/codeflow?start=47&length=31)

Should this be a warning rather than an error in an `unsafe` context?

Reply:
Yes, I think so